numpy/core/arrayprint.py

```diff
@@ -706,9 +706,9 @@ def _extendLine(s, line, word, line_width, next_line_prefix, legacy):
             needs_wrap = False
 
     if needs_wrap:
-        s += line.rstrip() + "\n"
+        s = ''.join((s, line.rstrip(), "\n"))
         line = next_line_prefix
-    line += word
+    line = ''.join((line, word))
```

Comment:
In particular, when joining only two strings like this you shouldn't find any performance increase at all

Reply:
Though memory consumption has decreased. Timings are still the same which I think is expected.

Reply:
This particular change should make memory worse not better